overlord/snapstate/handlers.go

```diff
@@ -929,11 +929,14 @@ func asyncRefreshOnSnapClose(st *state.State, refreshInfo *userclient.PendingSna
 		st.Lock()
 		defer st.Unlock()
 
+		monitoredSnaps := st.Cached("monitored-snaps").(map[string]bool)
 		delete(monitoredSnaps, refreshInfo.InstanceName)
+
 		if len(monitoredSnaps) == 0 {
-			monitoredSnaps = nil
```

Comment:
Maybe a small comment here like: `// important to set the cached value to nil directly and not monitoredSnaps as a nil map is different from a nil value` ? But maybe too complicated to explain in a single line? (`// I have discovered a truly marvelous explaination of this, which this margin is too narrow to contain`)